### ISSUE TITLE
[FR] Integrate GPT4ALL, open source language model

### ISSUE THREAD

**Issue by tio-trom** [AppFlowy-IO/AppFlowy#3544]
### Description

Gpt4all is an open source and totally free alternative to ChatGPT. See https://gpt4all.io/index.html 

It can do a lot. For one, you can replace the OpenAI tool with it, so you can write paragraphs with GPT4all. But even more awesome you can integrate the search and let GPT4all use the notes stored in the AppFlowy in order to search through them - see https://docs.gpt4all.io/gpt4all_chat.html#plugins LocalDocs Beta Plugin (Chat With Your Data)

### Impact

I have hundreds of notes, from personal notes, to a lot of code snippets and such. Imagine if I could be able to search for something specific or ask like "hey how to fix my website X SSH connection?" and the Gpt4all can see in my notes how I fixed that and tell me in a natural way. 

This could be the most mindblowing and useful tool for notes. Not just search, but inquiry of your notes with a language model.

It runs locally so there is no data collection and no OpenAI BS.

I think this could make AppFlowy the most amazing notes app in the world :)

**Comment by MayurSMahajan** [comment]
cc @yatendra2001

**Comment by yatendra2001** [comment]
Hey @tio-trom , this does sound like a freaking awesome idea. 

Btw we're already working on integrating multiple opensource models in a way that users have the power to choose whatever model they wanna choose. Searching through the notes can enhance productivity significantly.

cc: @appflowy

**Comment by tio-trom** [comment]
Mind you tho currently the search through documents will not limit itself to those documents. So I am not sure it can work as I was suggesting....

I am testing GPT4all currently and yeah....it is super slow but maybe this will improve with time and different models. It is definitely worth looking into this.

**Comment by yatendra2001** [comment]
yup exactly that's the crux, most gpt models aren't fast enough to go live into production. that's why we want to give this authority to users, they can choose whatever models they want to.
